matchms/Spectrum.py

```diff
@@ -72,30 +83,31 @@ def __init__(self,
             Array of intensities for the peaks
         metadata
             Dictionary with for example the scan number of precursor m/z.
+        metadata_harmonization : bool, optional
+            Set to False if default metadata filters should not be applied.
+            The default is True.
         """
-        if metadata is None:
-            self._metadata = {}
-        else:
-            self._metadata = metadata
-        self.peaks = Spikes(mz=mz, intensities=intensities)
+        self._metadata = Metadata(metadata)
+        if metadata_harmonization is True:
+            self._apply_metadata_harmonization()
+        self.peaks = Fragments(mz=mz, intensities=intensities)
         self.losses = None
 
     def __eq__(self, other):
         return \
             self.peaks == other.peaks and \
             self.losses == other.losses and \
-            self.__metadata_eq(other.metadata)
-
-    def __metadata_eq(self, other_metadata):
-        if self.metadata.keys() != other_metadata.keys():
-            return False
-        for i, value in enumerate(list(self.metadata.values())):
-            if isinstance(value, numpy.ndarray):
-                if not numpy.all(value == list(other_metadata.values())[i]):
-                    return False
-            elif value != list(other_metadata.values())[i]:
-                return False
-        return True
+            self._metadata == other._metadata
+
+    def _apply_metadata_harmonization(self):
+        metadata_filtered = _interpret_pepmass_metadata(self.metadata)
+        if metadata_filtered.get("ionmode") is not None:
+            metadata_filtered["ionmode"] = self.metadata.get("ionmode").lower()
+        metadata_filtered = _add_precursor_mz_metadata(metadata_filtered)
+        charge = metadata_filtered.get("charge")
+        if not isinstance(charge, int) and not _convert_charge_to_int(charge) is None:
+            metadata_filtered["charge"] = _convert_charge_to_int(charge)
+        self._metadata = Metadata(metadata_filtered)
```

Comment:
Or should it be used via the `add_retention_time` and `add_retention_index` methods in this part?

Reply:
Yes, I would add it here. I guess we could use the helper functions from add_retention.py and apply it here (or change the structure of add_retention a bit such that there is a helper function which only changes the metadata, that's what I did for `add_precursor_mz`).

Reply:
So the harmonization basically only transfers the value if it is present in one of the keys, without changing the actually stored value, am I right?

Reply:
We have a bit of a duplication here at the moment, because the PickyDict part will already to most of the key harmonization. So if we have data coming in with "retentiontime" or "PrecursorMass" then those keys will automaticall have been replaced by "retention_time" and "precursor_mz" even before this method is called.

What is happening here is the harmonization of the entries itself. So convert strings to numbers (e.g. charge: "2+" --> 2) or make ionmode entries lower-case.

Reply:
So converting strings to numbers is also already part of PickyDict?

Reply:
No, sorry maybe that wasn't very clear. PickDict only handles the dictionary keys. The string-to-number conversions are part of the functions in this part (e.g. `_convert_charge_to_int`).

Reply:
ah, so these are the default filters - that is how I implemented it now. I think making this distinction also makes sense: harmonization -> store value in right place, default_filter: make sure value is valid if it is present

The nice part is then that the default filters only have to check the canonical keyword